File: 151003/Mytnik/Lab5/Discussion/NoteEntity/Note.cs
using Discussion.Common;

namespace Discussion.NoteEntity
{
    public class Note(string content, string country) : AbstractEntity
    {
        public string Country { get; set; } = country;
        public int TweetId { get; set; }
        public string Content { get; set; } = content;

        public Note() : this(string.Empty, string.Empty) { }

        public Note(int id, int tweetId, string content) : this(content, string.Empty)
        {
            Id = id;
            TweetId = tweetId;
        }

        public Note(int id, int tweetId, string content, string country) : this(content, country)
        {
            Id = id;
            TweetId = tweetId;
        }
    }
}

File: 151001/Shcherba/lab4/discussion/src/main/java/com/example/discussion/model/Note.java
package com.example.discussion.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn;
import org.springframework.data.cassandra.core.mapping.Table;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(value = "tbl_note")
public class Note {
    @PrimaryKeyColumn(name = "country", type = PrimaryKeyType.PARTITIONED)
    String country;

    @PrimaryKeyColumn(name = "tweetid", ordinal = 0, type = PrimaryKeyType.CLUSTERED)
    int tweetId;

    @PrimaryKeyColumn(name = "id", ordinal = 1, type = PrimaryKeyType.CLUSTERED)
    int id;

    String content;
}
